WebContent/zip-ext.js

```diff
@@ -208,11 +208,165 @@
+				window.addEventListener("unload", close);
+				addInstanceDB(callback, onerror);
+			};
+		}
+
+		function addInstanceDB(callback, onerror) {
+			var request;
+			try {
+				request = indexedDB.open(dbName + "_" + instance, {temporary: true});
+			} catch (e) {
+				request = indexedDB.open(dbName + "_" + instance);
+			}
+			request.onerror = onerror;
+			request.onupgradeneeded = function (event) {
+				tempDB = event.target.result;
+				tempDB.createObjectStore("files", { autoIncrement: true });
+			};
+			request.onsuccess = function (event) {
+				callback();
+			};
+		}
+
+		function close() {
+			blobs = [];
+			indexedDB.deleteDatabase(dbName + "_" + instance);
+			window.removeEventListener("storage", pongDB);
+			window.removeEventListener("unload", pongDB);
+		}
+
+		function broadcastPingDB() {
+			sessionStorage.zipjs = Date.now();
+		}
+
+		function pongDB(event) {
+			if ((event !== undefined && event.key === "zipjs") && instance) {
+				db.transaction(["instances"], "readwrite")
+								.objectStore("instances")
+								.put(Date.now(), instance);
+			}
+		}
+
+		function cleanUpOldInstances() {
+			broadcastPingDB();
+			pongDB();
+			setTimeout(function () {
+				findOldInstances(10000);
+			}, 5000);
+		}
+
+		function findOldInstances(maxAge) {
+			var t = db.transaction(["instances"], "readwrite"),
+				instances = t.objectStore("instances"),
+				expiration = Date.now() - maxAge,
+				oldInstances = [];
+
+			instances.openCursor().onsuccess = function (event) {
+				var cursor = event.target.result;
+				if (cursor) {
+					if (cursor.value < expiration) {
+						oldInstances.push(cursor.key);
+						cursor.delete();
+					}
+					cursor.continue();
+				} else {
+					deleteOldInstances(oldInstances);
+				}
+			};
+
+			t.onerror = onerror;
+		}
+
+		function deleteOldInstances(oldInstances) {
+			var index = 0;
+
+			for (var i = 0; i < oldInstances.length; i++) {
+				indexedDB.deleteDatabase(dbName + "_" + oldInstances[i]);
+			}
+		}
+
+		function makeDBBackedBlob(blob, callback, onerror) {
+			function putEntry(blob) {
+				var t = tempDB.transaction(["files"], "readwrite"),
+					objectStore, request;
+				t.onerror = onerror;
+				objectStore = t.objectStore("files");
+				request = objectStore.put({instance: instance, data: blob});
+				request.onsuccess = function (event) {
+					getEntry(request.result);
+				};
+			}
+
+			function getEntry(key) {
+				var t = tempDB.transaction(["files"], "readonly"),
+					objectStore, request;
+				t.onerror = onerror;
+				objectStore = t.objectStore("files");
+				request = objectStore.get(key);
+				request.onsuccess = function (event) {
+					callback(request.result.data);
+				};
+			}
+
+			putEntry(blob);
+		}
+
+
+		function writeUint8Array(array, callback, onerror) {
+			var blob = new Blob([ appendABViewSupported ? array : array.buffer ]);
+
+			makeDBBackedBlob(blob, function (storedBlob) {
+					blobs.push(storedBlob);
+					callback();
+				}, onerror);
```

Comment:
This line and the two lines before: Fix the weird spacing by unindenting it one level (so that the last `}` is at the same level as `makeDBBackedBlob(blob, function() {`.